server/polar/event/repository.py

```diff
@@ -191,3 +273,211 @@ def get_by_pending_entries_for_meter_statement(
 
     def get_eager_options(self) -> Options:
         return (joinedload(Event.customer),)
+
+    async def list_with_closure_table(
+        self,
+        statement: Select[tuple[Event]],
+        limit: int,
+        page: int,
+        aggregate_costs: bool = False,
+    ) -> tuple[Sequence[Event], int]:
+        """
+        List events using closure table to get a correct children_count.
+        Optionally aggregates costs from descendants into user_metadata._cost.amount.
+        """
+        descendant_event = Event.__table__.alias("descendant_event")
```

Comment:
I think you can use `aliased` as a shortcut: https://docs.sqlalchemy.org/en/20/orm/queryguide/api.html#sqlalchemy.orm.aliased